### PR TITLE
feat: replace async/await with void because sync rules are supported

### PR DESCRIPTION
**Pull request by JounQin**
continue https://github.com/markuplint/markuplint/pull/157#issuecomment-827494898

### CODE REVIEW

File: packages/@markuplint/rules/src/attr-duplication/index.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { createRule } from '@markuplint/ml-core';
 
 export default createRule({
-	async verify({ document, report, t }) {
+	verify({ document, report, t }) {
 		const message = t('{0} is {1:c}', t('the {0}', 'attribute name'), 'duplicated');
-		await document.walkOn('Element', node => {
+		void document.walkOn('Element', node => {
```
Comment (JounQin) [inline packages/@markuplint/rules/src/attr-duplication/index.ts:6]: `void` is used to workaround for [`@typescript-eslint/no-floating-promises`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-floating-promises.md).

Comment (YusukeHirao) [reply packages/@markuplint/rules/src/attr-duplication/index.ts:6]: Nice!

If so, could you add to the rule for ESLint?

Comment (JounQin) [reply packages/@markuplint/rules/src/attr-duplication/index.ts:6]: The default rule is https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/no-floating-promises.ts#L55, but the CI is broken, I'll check ASAP.

**Comment by JounQin** [comment]
@YusukeHirao 

I'm a bit confused by the following:

https://github.com/markuplint/markuplint/blob/676df58faec6218c2d58dce3a75e0b8d29aa6ea9/packages/%40markuplint/ml-core/src/ml-dom/node/node.ts#L456-L458

**Comment by YusukeHirao** [comment]
@JounQin Sorry to bother you again.
Finally, I remember why I didn't remove await operators now.

It can't support sync, for it needs sequential execution.
Because the document object and elements include the settings of rules optimized from selectors. So it needs to create the rule context in the order, and it can't parallel execute them.
Therefore you may not remove await operators.

Due to the need for rule contexts, the process is complex a little bit.

And there is another reason. Because the function `report` is a side effect, not the return value. 

I should have recognized it in the failed test.

**Comment by JounQin** [comment]
Never mind.

**Pull request closed without merging**